source/deploy/mobile/secure-file-storage.rst

```diff
@@ -0,0 +1,165 @@
+
+- **Server-Side Access Controls:** Administrators can enforce policies through the System Console to restrict file downloads, sharing, and public link generation. Currently, policies are applied at the server level. For more details, see `Configuration Settings - File Sharing and Downloads <https://docs.mattermost.com/configure/site-configuration-settings.html#file-sharing-and-downloads>`_.
+
+- **Sandbox isolation:** As discussed earlier, Mattermost’s mobile apps store files in a sandboxed environment. This isolation ensures that even if a device is shared or compromised, other apps cannot access the cached files without explicit user action.
+
+- **Controlled file previews:** File previews (via QLPreviewController on **iOS** or open intent on **Android**) occur within the application’s controlled environment or delegate to OS-approved apps without duplicating sensitive data outside the sandbox.
+
+- **OS-Level encryption:** While files in the cache folder are stored unencrypted by default, they are protected at rest by OS-level encryption (iOS File Data Protection and Android Full Disk Encryption).
+
+By combining strict authentication, server-side controls, sandbox isolation, and secure file handling, Mattermost ensures that sensitive data remains confined to authorized users, with access and sharing tightly controlled by server policies, platform security features, and explicit user consent.
+
+Testing procedure to validate file isolation
+---------------------------------------------
+
+The playbook below aims to test and confirm that files stored in the Mattermost app’s private cache folder—where sensitive data like CUI may reside—are not accessible by unauthorized third-party applications. The tests are designed to show that while public directories, such as Downloads/ can be accessed, the private Mattermost cache remains isolated unless a file is explicitly shared by an authorized user.
+
+1. **Preparation**
+
+   - On **Android**:
+      - Install Adobe Acrobat Reader for PDF viewing.
+      - Install Files by Google as a file manager.
+      - Install Google Chrome as a system browser.
+   - On **iOS**:
+      - Install Adobe Acrobat Reader (from the App Store).
+      - Use the built-in Files app.
+      - Use Safari as the default browser.
+   - Any PDF viewer, file manager, or browser can be used for these steps.
+   
+2. **Control test: Verify public directory accessibility**
+
+- **Android:**
+  - Open Files by Google.
+  - Navigate to a known public directory (e.g., Downloads/).
+  - Confirm that files in this location are visible and can be opened.
+
+- **iOS:**
+  - Open the Files app.
+  - Browse to an accessible directory (e.g., iCloud Drive or “On My iPhone” public folders).
+  - Confirm these files are accessible.
+
+This step demonstrates that allowed directories can be accessed, setting a baseline for comparison.
```

Comment:
Fixing indentation so that the bullets are children of bullet 2

```suggestion
   - **Android:**
      - Open Files by Google.
      - Navigate to a known public directory (e.g., Downloads/).
      - Confirm that files in this location are visible and can be opened.

   - **iOS:**
      - Open the Files app.
      - Browse to an accessible directory (e.g., iCloud Drive or “On My iPhone” public folders).
      - Confirm these files are accessible.

   This step demonstrates that allowed directories can be accessed, setting a baseline for comparison.
```